dev_roadmap.md

```diff
@@ -0,0 +1,341 @@
+ There is also some potential for renaming, as some module names have underscores, which is unconventional.
+ 
+  #### Qtrl
+ *current package status*: integrated sub-package `qutip.control`
+ 
+ *planned package status*: family package `qtrl`
+ 
+ There are many OSS Python packages for quantum control optimisation. There are also many different algorithms. 
+ The current `control` integrated subpackage provides the GRAPE and CRAB algorithms. 
+ It is too ambitious for QuTiP to attempt (or want) to provide for all options.
+ Control optimisation has been deemed out of scope and hence these components will be separated out into a family package called Qtrl.
+ 
+ Potentially Qtrl may be replaced by separate packages for GRAPE and CRAB, based on the QuTiP Control Framework.
+ 
+ #### QIP
+ *current package status*: integrated sub-package `qutip.qip`
+ 
+ *planned package status*: optional sub-package `qutip.qip`
+ 
+&#x1F537; Is it really necessary for this to be a sub-package? It could just be a separate package.
+
+The QIP subpackage has been deemed out of scope (feature-wise). It also depends on `qutip.control` and hence would be out of scope for dependency reasons.
+A separate repository has already been made for qutip-qip.
+ 
+ #### qutip-symbolic
+ *current package status*: independent package `sympsi`
+ 
+ *planned package status*: family package `qutip-symbolic`
+ 
+ Long ago Robert Johansson and Eunjong Kim developed Sympsi. It is a fairly coomplete library for quantum computer algebra (symbolic computation). 
+ It is primarily a quantum wrapper for [Sympy](https://www.sympy.org).
+ 
+ It has fallen into unmaintained status. The latest version on the [sympsi repo](https://github.com/sympsi/sympsi) does not work with recent versions of Sympy.
+ Alex Pitchford has a [fork](https://github.com/ajgpitch/sympsi) that does 'work' with recent Sympy versions -- unit tests pass, and most examples work. 
+ However, some (important) examples fail, due to lack of respect for non-commuting operators in Sympy simplifcation functions (note this was true as of Nov 2019, may be fixed now).
+ 
+ There is a [not discussed with RJ & EK] plan to move this into the QuTiP family to allow the Admin Team to maintain, develop and promote it. 
+ The 'Sympsi' name is cute, but a little abstract, and qutip-symbolic is proposed as an alternative, as it is plainer and more distinct from Sympy.
+ 
+ ### Affilliated packages
+ 
+ #### qucontrol-krotov
+ https://github.com/qucontrol/krotov
+ 
+ A package for quantum control optimisation using Krotov, developed mainly by Michael Goerz.
+ 
+ Generally accepted by the Admin Team as well developed and maintained. A solid candiate for affilliation.
+ 
+ --------------------------------------------------------------
+
+ ## Development Projects
+ 
+ ### data layer abstraction 
+ &#x1F535; tag: dl-abs
+ 
+ *status*: development [pretty much] complete
```

Comment:
"pretty much" is a _little_ optimistic - there's still bits to do here in providing full control over how the data-layer dispatchers choose the most appropriate output type - but definitely the heaviest lifting is done.  I'm going to try and write the final design spec for the little remaining stuff this weekend.

Reply:
Thanks @jakelishman 
I will update to better reflect the actual status